From bbdd868a76386a9987b085dc14a759bdb13a4cd5 Mon Sep 17 00:00:00 2001
From: Arian <arian.kulmer@web.de>
Date: Tue, 15 Oct 2024 19:06:16 +0200
Subject: [PATCH 09/10] Force enable StartActivityWhenLocked

MIUI sends this with the intent, but on AOSP we don't get this parameter
---
 smali/com/android/camera/s2.smali | 65 ++-----------------------------
 1 file changed, 3 insertions(+), 62 deletions(-)

diff --git a/smali/com/android/camera/s2.smali b/smali/com/android/camera/s2.smali
index 874c246cd..b224d0c76 100644
--- a/smali/com/android/camera/s2.smali
+++ b/smali/com/android/camera/s2.smali
@@ -280,70 +280,11 @@
 .end method
 
 .method public static O(Landroid/content/Intent;)Z
-    .locals 3
-
-    const/4 v0, 0x0
-
-    if-eqz p0, :cond_1
-
-    const-string v1, "StartActivityWhenLocked"
-
-    invoke-virtual {p0, v1, v0}, Landroid/content/Intent;->getBooleanExtra(Ljava/lang/String;Z)Z
-
-    move-result v1
-
-    if-nez v1, :cond_0
-
-    const-string v1, "from_keyguard_shortcut"
-
-    invoke-virtual {p0, v1, v0}, Landroid/content/Intent;->getBooleanExtra(Ljava/lang/String;Z)Z
-
-    move-result v1
-
-    if-nez v1, :cond_0
-
-    const-string v1, "android.media.action.STILL_IMAGE_CAMERA_SECURE"
-
-    invoke-virtual {p0}, Landroid/content/Intent;->getAction()Ljava/lang/String;
-
-    move-result-object p0
-
-    invoke-virtual {v1, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
-
-    move-result p0
-
-    if-eqz p0, :cond_1
-
-    :cond_0
-    const/4 p0, 0x1
-
-    goto :goto_0
-
-    :cond_1
-    move p0, v0
-
-    :goto_0
-    new-instance v1, Ljava/lang/StringBuilder;
-
-    invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
-
-    const-string/jumbo v2, "start activity when locked: "
-
-    invoke-virtual {v1, v2}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v1, p0}, Ljava/lang/StringBuilder;->append(Z)Ljava/lang/StringBuilder;
-
-    invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
-
-    move-result-object v1
-
-    new-array v0, v0, [Ljava/lang/Object;
-
-    const-string v2, "CameraIntentManager"
+    .locals 1
 
-    invoke-static {v2, v1, v0}, Lcom/android/camera/log/Log;->d(Ljava/lang/String;Ljava/lang/String;[Ljava/lang/Object;)V
+    const/4 v0, 0x1
 
-    return p0
+    return v0
 .end method
 
 .method public static Q(Landroid/content/Intent;)Z
-- 
2.47.2

